### DISCUSSION TITLE
how query handles data/query across the entire app?

### DISCUSSION THREAD

**Discussion by aboveyunhai** [TanStack/query#3923] [Q&A] [answered]
https://tanstack.com/query/v4/docs/guides/does-this-replace-client-state
Hi community, I am new to this query lib, had used `apollo-graphql` and `use-swr` before.
I read through the client state page but I'm still unclear about the following scenario.
Assume I have infinite query to retrieve posts in the main page like the below:

post 1
post 2
Load more button 

then I click `Load more` so the layout becomes the following:

post 1
post 2
post 3
Load more button 

then I click post 2 and redirect to the post 2 page and click route back to the main page.

**Question 1:**

will all the data gone and query does the refetch ? or it will still keep `the layout with post 3`.

Basically, the client state across the application before I click refresh by browser.

**Question 2:**
basically an elaboration of question 1, if it keeps `the layout with post 3`, it means data is persisted somewhere. 
Then if I called the `post infinite query` in the `post 2` page, will it change the layout of the main page and become `post 1, 2 ,3, 4` after I route back?

**Comment by TkDodo** [comment, marked as answer]
> Question 1:
> will all the data gone and query does the refetch ? or it will still keep the layout with post 3.

assuming a client side navigation and that each page uses the same QueryClient, the data will still be there.

> Question 2:

stale queries will be refetched in the background. Assuming  a staleTime of zero (the default), if you navigate back to the list and the list mounts, a refetch will happen.

If you make a mutation, you can manually invalidate matching queries with `queryClient.invalidateQueries` to force a refetch right away.

**Reply by aboveyunhai** [reply]
@TkDodo  thx for your answer. For stale query, can I manually disable the refetch mechanism on purpose? Hence it will stay forever until users do refetch by themselves regardless the client side navigation.

**Reply by TkDodo** [reply]
that's what `staleTime: Infinity` would do
